pkg/component/vpnseedserver/vpn_seed_server.go

```diff
@@ -868,20 +881,34 @@ func (v *vpnSeedServer) emptyEnvoyFilter() *networkingv1alpha3.EnvoyFilter {
 	return &networkingv1alpha3.EnvoyFilter{ObjectMeta: metav1.ObjectMeta{Name: v.namespace + "-vpn", Namespace: v.istioNamespaceFunc()}}
 }
 
+//TODO: unused function?
 func getLabels() map[string]string {
 	return map[string]string{
 		v1beta1constants.GardenRole: v1beta1constants.GardenRoleControlPlane,
 		v1beta1constants.LabelApp:   DeploymentName,
 	}
 }
 
-var envoyConfig = `static_resources:
+func (v *vpnSeedServer) getEnvoyConfig() string {
+	var (
+		listenAddress = "0.0.0.0"
+		// we don't want to use AUTO for single-stack clusters as it causes an unnecessary failed lookup
+		// ref https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/cluster/v3/cluster.proto#enum-config-cluster-v3-cluster-dnslookupfamily
+		dnsLookupFamily = "V4_ONLY"
+	)
+
+	if gardencorev1beta1.IsIPv6SingleStack(v.values.Network.IPFamilies) {
+		listenAddress = "::"
+		dnsLookupFamily = "V6_ONLY"
+	}
+
+	return `static_resources:
   listeners:
   - name: listener_0
     address:
       socket_address:
         protocol: TCP
-        address: 0.0.0.0
+        address: ` + fmt.Sprintf("%q", listenAddress) + `
```

Comment:
why not
```suggestion
        address: "` + listenAddress + `"
```
?

same below

Reply:
https://pkg.go.dev/fmt#pkg-overview

```
%q	a single-quoted character literal safely escaped with Go syntax.
```